compute_endpoint/globus_compute_endpoint/cli.py

```diff
@@ -870,35 +867,47 @@ def delete_endpoint(
     "compress",
     default=False,
     is_flag=True,
-    help="Save the output to a Gzip-compressed file.",
+    help=(
+        "Generate a Gzip-compressed file only, and skip printing diagnostic "
+        "results to the console."
+    ),
 )
 @click.option(
+    "-k",
     "--log-kb",
     default=5120,
     help=(
         "Specify the number of kilobytes (KB) to read from log files."
         " Defaults to 5,120 KB (5 MB)."
     ),
 )
+@click.option(
+    "-e",
+    "--endpoint-uuid",
+    default=None,
+    help=(
+        "Test an endpoint by registering a sample function and sending "
+        "a task to it using the newly registered function.  An endpoint "
+        "UUID is required."
+    ),
+)
 @click.help_option("-h", "--help")
-def self_diagnostic(compress: bool, log_kb: int):
+def self_diagnostic(compress, log_kb: int, endpoint_uuid: str):
     """Run several diagnostic commands to help identify issues.
 
     This may produce a large amount of output.
-    """
-    log_bytes = log_kb * 1024
+    Note that this functionality has been migrated to the SDK, leaving only
+    a wrapper here.
 
-    if not compress:
-        run_self_diagnostic(log_bytes=log_bytes)
-    else:
-        current_date = datetime.now().strftime("%Y-%m-%d")
-        filename = f"globus_compute_diagnostic_{current_date}.txt.gz"
+    The -z and --log-kb arguments have been preserved from the previous
+    command, though the diagnostic always generates a gzipped file now,
+    with printing to console being optional.
 
-        with gzip.open(filename, "wb") as f:
-            with contextlib.redirect_stdout(f):  # type: ignore[type-var]
-                run_self_diagnostic(log_bytes=log_bytes)
-
-        click.echo(f"Successfully created {filename}")
+    TODO should we add a deprecation message here? ie.
+        The endpoint specific self-diagnostic command has been deprecated.
+        Please use the `globus-compute-diagnostic` command instead.
```

Comment:
Yes; and perhaps at the same time, internally simply have this one call the `globus-compute-diagnostic`?

Reply:
I use a wrapper to keep the previous arguments the same, and also was wary of click and argparse fighting over arguments.

Left some code commented out for a new version that calls the SDK version after shifting the arguments by one (``globus-compute-endpoint self-diagnostic x y z`` -> ``globus-compute-diagnostic x y z``)

Reply:
Added deprecation warning, globus-compute-endpoint self-diagnostic now redirects.

```
$ ge self-diagnostic --help
Usage: globus-compute-endpoint self-diagnostic [OPTIONS]

  Run several diagnostic commands to help identify issues.

  This may produce a large amount of output.

Options:
  -z, --gzip        Save the output to a Gzip-compressed file.
  --log-kb INTEGER  Specify the number of kilobytes (KB) to read from log
                    files. Defaults to 5,120 KB (5 MB).
  -h, --help        Show this message and exit.
  ```

Reply:
Added a print to stderr instead of the warnings.warn, which was a standard deprecation method?  This way it is more visible.

Reply:
> Added a print to stderr instead of the warnings.warn, which was a standard deprecation method?

Yes; the distinction is that the Python ecosystem typically views deprecation notices like this as "for library implementors and maintainers."  By default, then, Python won't emit those notices.  Hence the suggestion that makes this more visible for our users.